adapters/sharethrough/sharethrough.go

```diff
@@ -1,62 +1,109 @@
 	"github.com/prebid/prebid-server/adapters"
 	"github.com/prebid/prebid-server/config"
 	"github.com/prebid/prebid-server/errortypes"
 	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/prebid/prebid-server/version"
 )
 
-const supplyId = "FGMrCMMc"
-const strVersion = 8
+var adapterVersion = "10.0"
+
+type adapter struct {
+	endpoint string
+}
 
 // Builder builds a new instance of the Sharethrough adapter for the given bidder with the given config.
 func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
-	bidder := &SharethroughAdapter{
-		AdServer: StrOpenRTBTranslator{
-			UriHelper: StrUriHelper{BaseURI: config.Endpoint, Clock: Clock{}},
-			Util:      Util{Clock: Clock{}},
-			UserAgentParsers: UserAgentParsers{
-				ChromeVersion:    regexp.MustCompile(`Chrome\/(?P<ChromeVersion>\d+)`),
-				ChromeiOSVersion: regexp.MustCompile(`CriOS\/(?P<chromeiOSVersion>\d+)`),
-				SafariVersion:    regexp.MustCompile(`Version\/(?P<safariVersion>\d+)`),
-			},
-		},
+	bidder := &adapter{
+		endpoint: config.Endpoint,
 	}
 	return bidder, nil
 }
 
-type SharethroughAdapter struct {
-	AdServer StrOpenRTBInterface
-}
+func (a adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var requests []*adapters.RequestData
+	var errors []error
 
-func (a SharethroughAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
-	var reqs []*adapters.RequestData
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	modifiableSource := openrtb2.Source{}
+	if request.Source != nil {
+		modifiableSource = *request.Source
+	}
+	var sourceExt map[string]interface{}
+	if err := json.Unmarshal(modifiableSource.Ext, &sourceExt); err == nil {
+		sourceExt["str"] = adapterVersion
+		sourceExt["version"] = version.Ver
+	} else {
+		sourceExt = map[string]interface{}{"str": adapterVersion, "version": version.Ver}
+	}
 
-	if request.Site == nil {
-		return nil, []error{fmt.Errorf("request must include a site; in-app placements are not supported")}
+	var err error
+	if modifiableSource.Ext, err = json.Marshal(sourceExt); err != nil {
+		errors = append(errors, err)
 	}
-	var domain = Util{}.parseDomain(request.Site.Page)
+	request.Source = &modifiableSource
+
+	requestCopy := *request
+	for _, imp := range request.Imp {
+		// Extract Sharethrough Params
+		var strImpExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &strImpExt); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+		var strImpParams openrtb_ext.ExtImpSharethrough
+		if err := json.Unmarshal(strImpExt.Bidder, &strImpParams); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		// Convert Floor into USD
+		if imp.BidFloor > 0 && strings.EqualFold(imp.BidFloorCur, "USD") {
```

Comment:
There is no test coverage within this condition. Is that possible to add? .. or is our json test framework not up for the task?

Reply:
I haven't found a way to mock the bid floor conversion, also I was ok with not testing it since it's a copy paste from the Prebid doc. If there's a way and some examples of testing this via the json framework, I'm happy to add a test.

Reply:
I don't think we have an easy way to do that. I'm fine with skipping test coverage for now. We may add in tests for this adapter, and the others, when we add per-test definable currency conversion rates.